akka-http-core/src/main/java/akka/http/javadsl/model/HttpHeader.java

```diff
@@ -44,4 +44,22 @@
      * Returns true if and only if the header is to be rendered in responses.
      */
     public abstract boolean renderInResponses();
+
+    /**
+     * Attempts to parse the given header name and value string into a header model instance.
+     *
+     * @throws IllegalArgumentException if parsing is unsuccessful.
+     */
+    public static HttpHeader parse(String name, String value) {
+      final akka.http.scaladsl.model.HttpHeader.ParsingResult result =
+        akka.http.scaladsl.model.HttpHeader.parse(name, value,
+          akka.http.impl.model.parser.HeaderParser$.MODULE$.DefaultSettings());
+
+      if (result instanceof akka.http.scaladsl.model.HttpHeader$ParsingResult$Ok) {
+        return ((akka.http.scaladsl.model.HttpHeader$ParsingResult$Ok) result).header();
+      }
+      else {
+        throw new IllegalArgumentException("Unable to parse header [" + name + "] with value [" + value + "]");
```

Comment:
I wonder if safe to include the value here hm... If it's some auth header etc?

Reply:
Good point. Remove header name and value from the exception message.